### PR TITLE
Install Qt5 dlls

### PR DESCRIPTION
**Pull request by mathturtle**
This gets the superbuild to install the Qt5 dlls into the package on Windows.  Probably not the "best" way to do this, but it works.

### CODE REVIEW

**Comment by mathturtle** [comment]
@cryos @mathstuf

File: projects/win32/qt.use.system.cmake [hunk 1]
```diff
@@ -6,37 +6,37 @@ option(PACKAGE_SYSTEM_QT
...
+  Gui
+  Help
+  Network
+  Widgets
+  Sql
+  Test
+  UiTools
+  Xml)
+find_package(Qt5 COMPONENTS ${tomvizQt5Libs})
+list(APPEND tomvizQt5Libs QWindowsIntegrationPlugin)
```
Comment (mathstuf) [inline projects/win32/qt.use.system.cmake]: `Qt5::QWindowsIntegrationPlugin` isn't a target, is it?

File: projects/win32/qt.use.system.cmake [hunk 2]
```diff
@@ -6,37 +6,37 @@ option(PACKAGE_SYSTEM_QT
...
+file(GENERATE OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/cpack/qt5_bundle.cmake"
+  CONTENT "${qt5_libs_file_string}")
+install(CODE "
+set(CMAKE_MODULE_PATH \"${CMAKE_CURRENT_BINARY_DIR}/cpack\" \${CMAKE_MODULE_PATH})
+include(qt5_bundle)
+list(REMOVE_DUPLICATES qt5_libdirs)
+foreach(dir IN LISTS qt5_libdirs)
+  file(GLOB qt_libs \"\${dir}/*.dll\")
+  foreach (lib IN LISTS qt_libs)
+    if(NOT \"\${lib}\" MATCHES \"d.dll\$\")
```
Comment (mathstuf) [inline projects/win32/qt.use.system.cmake]: No need to dereference `lib` her:

```cmake
if (NOT lib MATCHES \"d.dll\$\")
```

File: projects/win32/qt.use.system.cmake [hunk 3]
```diff
@@ -6,37 +6,37 @@ option(PACKAGE_SYSTEM_QT
...
+      file(INSTALL \${lib}
+        DESTINATION \"\$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/bin\"
```
Comment (mathstuf) [inline projects/win32/qt.use.system.cmake]: Can just do `\"bin\"` here; relative paths get the `DESTDIR` and `CMAKE_INSTALL_PREFIX` automatically.

**Comment by cryos** [comment]
@mathstuf just confirming - ParaView currently has no support/code to package Qt 5? I want to be certain we are not repeating work I was under the impression had been done last year. I don't know the fixupbundle.py code very well, but that is also failing for Mac OS X builds, resulting in no dependent libraries being packaged in our bundle.

**Comment by mathstuf** [comment]
I don't see any changes specific to Qt5. Syncing fixup_bundle may help, but I was mostly stuck on getting Windows to build at all before looking at packaging problems.

**Comment by cryos** [comment]
OK, good to know. @mathturtle I need to leave, and will merge this branch later if you don't. I am done experimenting with the mac build for now, the following was present so it is clear that it found the main executable still.

1: ------------------------------------------------------------
1: Found executables : 
1:     tomviz.app/Contents/MacOS/tomviz

There was no output suggesting it found a single library, none indicating it didn't either... It appears to be running to completion, but is not packaging dependent libraries as expected for the Mac. I don't know that code, it seems very verbose in listing all libraries and I was surprised not to see any output listing discovered dependent libraries.

File: projects/win32/qt.use.system.cmake (hunk 1 above)
Comment (mathturtle) [inline projects/win32/qt.use.system.cmake]: Apparently you need this plugin on Windows.  @cryos pointed it out in the script to bundle avagadro: https://github.com/OpenChemistry/avogadroapp/blob/master/cmake/DeployQt5.cmake#L324

File: projects/win32/qt.use.system.cmake (hunk 3 above)
Comment (mathturtle) [inline projects/win32/qt.use.system.cmake]: Even inside an `install(CODE ...)`?

**Comment by mathturtle** [comment]
Merging with one minor change from @mathstuf's comments.  I'm leaving the other two as-is for now.

@cryos I'm done with bigmac for now, I found a typo in fixup_bundle that was causing more errors.  But now something is pulling in XCTest.framework which fails to be found.  I don't know what is pulling in XCode's test framework though, maybe Qt?  We shouldn't need that, so we may have to look into our dependencies.

File: projects/win32/qt.use.system.cmake [hunk 4]
```diff
@@ -6,37 +6,37 @@ option(PACKAGE_SYSTEM_QT
...
-          FILES_MATCHING REGEX "^.*d4.dll$" EXCLUDE
-          PATTERN "*.dll")
-endif()
+set(tomvizQt5Libs
+  Core
+  Gui
+  Help
+  Network
+  Widgets
+  Sql
```
Comment (cryos) [inline projects/win32/qt.use.system.cmake]: I doubt we need Sql or Test, UiTools looks unlikely but possible. If this is the same on Mac I could see Test pulling in XCode testing dependencies.

**Comment by mathturtle** [comment]
Sql I added because it complained when I tried to run it.  But the way I'm doing this now we only need Core anyway since we just need to find the directory to copy dlls out of.  I was initially trying to copy only the necessary files, but it was taking too long to figure out what they were.

Test and most of the others were copied from tomviz's CMake code where it links in the Qt modules.  Taking it out seems to make the tomviz build fail.  Why does tomviz need it?

**Comment by cryos** [comment]
Tomviz inherits its dependencies from the ParaView libraries on the whole, but the way modern CMake functions it adds warnings when you link to libraries from the imported targets that are not defined in that scope. There is no use of the Qt SQL or test modules in tomviz, if either are required they are coming from the ParaView libraries we link to.

File: projects/win32/qt.use.system.cmake (hunk 3 above)
Comment (mathstuf) [inline projects/win32/qt.use.system.cmake]: Hmm. Not sure.

**Pull request merged**